R/thumbnail.R

```diff
@@ -0,0 +1,233 @@
+#' Get the Content Thumbnail
+#'
+#' `get_thumbnail` saves the content thumbnail to the given path (default: temp file).
+#' `delete_thumbnail` removes the thumbnail (optionally saving to the given path)
+#' `has_thumbnail` returns whether the content has an thumbnail
+#'
+#' @param content A content object
+#' @param path optional. The path to the image on disk
+#'
+#' @rdname get_thumbnail
+#' @family content functions
+#' @export
+get_thumbnail <- function(content, path = NULL) {
+  validate_R6_class(content, "Content")
+  guid <- content$get_content()$guid
+
+  con <- content$get_connect()
+
+  # Connect 2024.09.0 introduced public endpoints for content thumbnails. We
+  # prefer those, falling back to the unversioned endpoints if unavailable.
+  res <- con$GET(
+    url = con$server_url("content", guid, "__thumbnail__"),
+    parser = NULL
```

Comment:
If the goal is to end up with the content written to a file, why not use `httr::write_disk()` here?

Reply:
Closing the loop here: because we want to get the result before deciding the file path, so we can determine the path to write to (see https://github.com/rstudio/connectapi/pull/303#discussion_r1775910845).